cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -262,6 +282,16 @@ func (p *pod) CreateTask(ctx context.Context, req *task.CreateTaskRequest, s *sp
 		}
 	}()
 
+	templateID := oci.ParseAnnotationsTemplateID(ctx, s)
+	saveAsTemplate := oci.ParseAnnotationsSaveAsTemplate(ctx, s)
+	if templateID != "" && saveAsTemplate {
+		return nil, errors.Wrapf(errdefs.ErrInvalidArgument, "templateID and save as template flags can not be passed in the same request.")
+	}
+
+	if (saveAsTemplate || templateID != "") && (p.host == nil || p.host.OS() != "windows") {
+		return nil, errors.Wrapf(errdefs.ErrInvalidArgument, "Save as template and creating clones is only available for WCOW.")
+	}
+
```

Comment:
nit: maybe move this to a function called something like `validateTemplateAnnotations`? What do you think?